scripts/main.py

```diff
@@ -316,118 +316,119 @@ def parse_arguments():
             cfg.memory_backend = chosen
 
 
-# TODO: fill in llm values here
-check_openai_api_key()
-parse_arguments()
-logger.set_level(logging.DEBUG if cfg.debug_mode else logging.INFO)
-ai_name = ""
-prompt = construct_prompt()
-# print(prompt)
-# Initialize variables
-full_message_history = []
-result = None
-next_action_count = 0
-# Make a constant:
-user_input = "Determine which next command to use, and respond using the format specified above:"
-
-# Initialize memory and make sure it is empty.
-# this is particularly important for indexing and referencing pinecone memory
-memory = get_memory(cfg, init=True)
-print('Using memory of type: ' + memory.__class__.__name__)
-
-# Interaction Loop
-while True:
-    # Send message to AI, get response
-    with Spinner("Thinking... "):
-        assistant_reply = chat.chat_with_ai(
-            prompt,
-            user_input,
-            full_message_history,
-            memory,
-            cfg.fast_token_limit) # TODO: This hardcodes the model to use GPT3.5. Make this an argument
-
-    # Print Assistant thoughts
-    print_assistant_thoughts(assistant_reply)
-
-    # Get command name and arguments
-    try:
-        command_name, arguments = cmd.get_command(attempt_to_fix_json_by_finding_outermost_brackets(assistant_reply))
-        if cfg.speak_mode:
-            speak.say_text(f"I want to execute {command_name}")
-    except Exception as e:
-        logger.error("Error: \n", str(e))
-
-    if not cfg.continuous_mode and next_action_count == 0:
-        ### GET USER AUTHORIZATION TO EXECUTE COMMAND ###
-        # Get key press: Prompt the user to press enter to continue or escape
-        # to exit
-        user_input = ""
-        logger.typewriter_log(
-            "NEXT ACTION: ",
-            Fore.CYAN,
-            f"COMMAND = {Fore.CYAN}{command_name}{Style.RESET_ALL}  ARGUMENTS = {Fore.CYAN}{arguments}{Style.RESET_ALL}")
-        print(
-            f"Enter 'y' to authorise command, 'y -N' to run N continuous commands, 'n' to exit program, or enter feedback for {ai_name}...",
-            flush=True)
-        while True:
-            console_input = utils.clean_input(Fore.MAGENTA + "Input:" + Style.RESET_ALL)
-            if console_input.lower().rstrip() == "y":
-                user_input = "GENERATE NEXT COMMAND JSON"
-                break
-            elif console_input.lower().startswith("y -"):
-                try:
-                    next_action_count = abs(int(console_input.split(" ")[1]))
+if __name__ == "__main__":
```

Comment:
Why not have this as a `def main()`: function instead? Then we can have:
```
if __name__ == "__main__":
    main()
```

This will open us up to then do more robust kickoff using something like [typer](https://typer.tiangolo.com/).